File: Backend/src/models/user_roles.py
# Backend/src/models/user_roles.py
# Many to Many relationship between users and roles

# Author: Valentin Haas, 2024

# Library imports
from sqlalchemy import Column, Integer
from sqlalchemy.ext.declarative import declarative_base

# Constants
BASE = declarative_base()


class UserRolesTable(BASE):
    """Model to track the relationship between users and roles."""

    __tablename__ = "UserRoles"

    User = Column(Integer, primary_key=True, foreign_key="Users.Id")
    """The user in the relationship."""

    Role = Column(Integer, primary_key=True, foreign_key="Roles.Id")
    """The role in the relationship."""

File: Backend/src/models/userroles.go
// Backend/src/models/userroles.go
// Many to Many relationship between users and roles

// Author: Valentin Haas, 2025
package models

import (
	"gorm.io/gorm"
)

// UserRole tracks the relationship between users and roles.
type UserRole struct {
	gorm.Model      // Provides ID, CreatedAt, UpdatedAt, DeletedAt fields
	User       User // Referenced User Object
	UserID     uint `gorm:"primaryKey"` // The user in the relationship (foreign key reference to User.ID).
	Role       Role // Referenced Role Object
	RoleID     uint `gorm:"primaryKey"` // The role in the relationship (foreign key reference to Role.ID).
}
